test/e2e/clusterctl_upgrade.go

```diff
@@ -694,8 +739,14 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 			By("Running PreCleanupManagementCluster steps against the management cluster")
 			input.PreCleanupManagementCluster(managementClusterProxy)
 		}
+
 		// Dumps all the resources in the spec namespace, then cleanups the cluster object and the spec namespace itself.
-		framework.DumpSpecResourcesAndCleanup(ctx, specName, input.BootstrapClusterProxy, input.ArtifactFolder, managementClusterNamespace, managementClusterCancelWatches, managementClusterResources.Cluster, input.E2EConfig.GetIntervals, input.SkipCleanup)
+		if input.UseKindForManagementCluster {
+			managementClusterProxy.Dispose(ctx)
```

Comment:
Would it make sense to call DumpAllResourcesAndLogs? (basically the same as DumpSpecResourcesAndCleanup with skipCleanup set to true)

Reply:
No.
DumpSpecResourcesAndLogs/Cleanup collects info from the bootstrap cluster, but since we are using another kind cluster, there is nothing about this test in the bootstrap cluster.

Instead, we are already collecting info from the kind cluster a few lines above